tests/python/relay/test_dataflow_pattern.py

```diff
@@ -601,6 +600,64 @@ def test_match_fake_diamond():
     assert not diamond.match(out)
 
 
+def test_at_most_one_parent():
```

Comment:
Ah, right. This test wast to confirm my understanding of the code, there's no harm leaving it even though it passes in main. The second test seems unnecessary so removed, sorry about that. None of these tests exercise the let-bound var dataflow changes in this PR since you can really only observe that at the lower IndexedGraph level. But the new c++ unit test does exercise all those changes.